### PR TITLE
Feature/header

### PR DESCRIPTION
**Pull request by whaeck**
The NJOY header and 2.0.x headers are now fully supported

### CODE REVIEW

File: python/src/Header201.python.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    python::init< const std::string&, const std::string&,
+                  double, double, const std::string&,
+                  const std::vector< std::string >& >(),
+    python::arg( "szaid" ), python::arg( "source" ), python::arg( "awr" ),
+    python::arg( "temperature" ), python::arg( "date" ),
+    python::arg( "comments" ),
+    "Initialise the component\n\n"
+    "Arguments:\n"
+    "    self          the header\n"
+    "    szaid         the szaid (max 10 characters)\n"
```
Comment (cjosey) [inline python/src/Header201.python.cpp:40]: I think the szaid in the new format is allowed to be 24 characters.

File: src/ACEtk/Table/Header201.hpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ *              SZAID (24 characters), blank space,
+ *              SRC (evaluation source, 24 characters)
+ *    - line 2: AW (atomic mass ratio, 12 characters), blank space,
+ *              TZ (temperature in MeV, 12 characters), blank space,
+ *              date (10 characters), blank space, N (number of comment lines,
+ *              4 characters)
+ *    - line 3 to N+2: comment lines (the last two lines are always the normal
+ *                     header)
+ */
+class Header201 {
```
Comment (cjosey) [inline src/ACEtk/Table/Header201.hpp:29]: To be honest, I am unclear why there are separate Header and Header201 classes. The data they contain is the same (201 being a superset) with perhaps the sole exception of the ZAID. Further, Lib80x files have both headers.

**Review by cjosey — COMMENTED** [review]
I think that I'm going to pause review of this one for further discussion in person and go on to the next few for now.

File: python/src/Header201.python.cpp (hunk 1 above)
Comment (whaeck) [reply python/src/Header201.python.cpp:40]: Yes.

File: src/ACEtk/Table/Header201.hpp (hunk 2 above)
Comment (whaeck) [reply src/ACEtk/Table/Header201.hpp:29]: By convention, they do but the last two lines being the old header was never "standardised" so they do contain different information (I think the title is no present in the 201 header).

**Review by cjosey — DISMISSED** [review]
We have discussed these changes, and improvements will be made in a future PR once the history is consistent.

**Review by cjosey — APPROVED** [review]

**Pull request merged**